internal/entities/properties/service/service.go

```diff
@@ -55,6 +56,10 @@ const (
 
 // PropertiesService is the interface for the properties service
 type PropertiesService interface {
+	// EntityWithPropertiesAsProto calls the provider to convert the entity with properties to the appropriate proto message
+	EntityWithPropertiesAsProto(
```

Comment:
At first I wanted to add this as a function to models instead of the service, but having it as a method makes testing easier as we generate mocks for the PropertiesService interface